elm/src/main/scala/io/scalac/elm/state/ElmWallet.scala

```diff
@@ -118,8 +118,8 @@ case class ElmWallet(secret: PrivateKey25519 = generateSecret(),
         val signature = Signature25519(Curve25519.sign(secret.privKeyBytes, out.bytes))
         TxInput(out.id, signature)
       }
-      val change = TxOutput(foundSum - requiredAmount, secret.publicImage)
-      val toRecipient = TxOutput(amount, secret.publicImage)
+      val change = TxOutput(foundSum - requiredAmount, to)
+      val toRecipient = TxOutput(amount, to)
```

Comment:
`toRecipient` is definitely a necessary bugfix, but `change` needs to stay as it was. Change goes back to the payer.